torch/_dynamo/trace_rules.py

```diff
@@ -2969,23 +2971,59 @@ def is_numpy(obj) -> bool:
     return isinstance(obj, (np.ndarray, np.generic)) or id(obj) in _numpy_function_ids
 
 
+"""
+Main entry point for looking up the trace rule (the Dynamo variable) for a given callable object.
+"""
+
+
+def lookup_callable(obj):
```

Comment:
callable can be anything with ```__call__``` implementation, and we use ```id()``` based match in Dynamo, so I make this a separate lookup function. ```trace_rules.lookup``` handles function type only. This would make ```VariableBuilder._wrap``` dispatch more clear logically.